packages/server/src/arpa_reporter/routes/exports.js

```diff
@@ -1,11 +1,50 @@
 const express = require('express');
 
 const router = express.Router();
+const { HeadObjectCommand, GetObjectCommand } = require('@aws-sdk/client-s3');
 const _ = require('lodash');
 
-const { requireUser } = require('../../lib/access-helpers');
+const aws = require('../../lib/gost-aws');
+const { requireUser, getAdminAuthInfo } = require('../../lib/access-helpers');
 const arpa = require('../services/generate-arpa-report');
 const { getReportingPeriodID, getReportingPeriod } = require('../db/reporting-periods');
+const { useTenantId, useUser } = require('../use-request');
+
+router.get('/:tenantId/:periodId/:filename', async (req, res) => {
+    let user;
+    try {
+        const info = await getAdminAuthInfo(req);
+        user = info.user;
+        if (user.tenant_id !== Number(req.params.tenantId)) {
+            throw new Error('Unauthorized');
+        }
+    } catch (error) {
+        res.redirect(encodeURI(`${process.env.WEBSITE_DOMAIN}/arpa_reporter/login?redirect_to=/api/exports/${req.params.tenantId}/${req.params.periodId}/${req.params.filename}&message=Please login to visit the link.`));
+        return;
+    }
+
+    const s3 = aws.getS3Client();
+    const Key = `${user.tenant_id}/${req.params.periodId}/${req.params.filename}`;
+    const baseParams = { Bucket: process.env.AUDIT_REPORT_BUCKET, Key };
+
+    try {
+        await s3.send(new HeadObjectCommand(baseParams));
+    } catch (error) {
+        console.log(error);
+        res.redirect(encodeURI(`${process.env.WEBSITE_DOMAIN}/arpa_reporter?alert_text=The treasury report you requested has expired. Please try again by clicking the 'Send Treasury Report By Email'.&alert_level=err`));
```

Comment:
Will this work? I think this will redirect, but not add the `async=True` param, which would mean the button won't show up.